File: DoItTest.Domain/Tests/TestItems/NumberFieldItem.cs
using DoItTest.Domain.Tests.TestItems.AnswerOptions;

namespace DoItTest.Domain.Tests.TestItems
{

    public class NumberFieldItem : TestItem
    {
        public NumberAnswerOption AnswerOption { get; }

        public NumberFieldItem(Guid id, Guid testId, TestItemType type,
            String question, String? imageBase64, NumberAnswerOption answerOption) 
            : base(id, testId, type, question, imageBase64)
        {
            AnswerOption = answerOption;
        }
    }
}

File: DoItTest/ClientApp/src/domain/tests/items/numberFieldItem.ts
import { mapNumericAnswerOption, NumericAnswerOption } from "./answerOption/numericAnswerOption";
import { TestItem } from "./testItem";
import { TestItemType } from "./testItemType";

export class NumberFieldItem extends TestItem {
    constructor(
        public id: string,
        public testId: string,
        public type: TestItemType,
        public question: string,
        public imageBase64: string | null,
        public answerOption: NumericAnswerOption
    ) {
        super(id, testId, type, question, imageBase64)
    }
}

export function mapToNumericFieldItem(value: any): NumberFieldItem {
    const answerOption = mapNumericAnswerOption(value.answerOption)

    return new NumberFieldItem(value.id, value.testId, value.type, value.question, value.imageBase64, answerOption)
}